app/controllers/annotations_controller.rb

```diff
@@ -233,6 +235,13 @@ def create
       @annotationOut['canvas'] = setMultipleCanvas
     end
 
+    if checkAuthorization(request.headers['Authorization'], @annotationOut['canvas']) == false
```

Comment:
In general `annotation['canvas']` or `annotation['on']['full']` won't contain a correct canvas ID when the target is not a canvas but another annotation. You should call `getTargetingAnnosCanvas(annotation)` to get the canvas ID, where the annotation object's `on` attribute is already present.

Reply:
I see that this needs the Annotation object to be passed to it. I will modify it to use that.